### PR TITLE
[DS-998] Spending By Category Visualization

### PR DESCRIPTION
**Pull request by ejgullo**
- Added 'Spending By' data to json
- Created sliced/diced treemap containing spending by information, displayed on homepage
- Added hover states using existing tooltip styling, moved arrow to top.  Description data is currently TBD, dummy info is in place.
- On tablet screen sizes and below, added a line label highlighting the top 75% of spending
- SCSS updates from @dfrasier

### CODE REVIEW

File: src/js/components/homepage/visualizations/categoryMap/CategoryMap.jsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        this.buildTree(this.props.breakdown, this.props.colors, this.state.category);
+    }
+
+    render() {
+        let tooltip = '';
+        let line = null;
+        if (this.state.windowWidth < 768) {
+            line = <BudgetLine />;
+        }
+        if (this.state.category !== 'none') {
```
Comment (mab6bf) [inline src/js/components/homepage/visualizations/categoryMap/CategoryMap.jsx]: Can we move this into it's own function? Something like `const tooltip = this.generateTooltip()`, where you'd return null if `this.state.category !== 'none'`

File: src/_scss/components/visualizations/tooltip/_arrow.scss [hunk 2]
```diff
@@ -26,4 +26,12 @@
             transform: rotate(225deg);
         }
     }
+    &.top {
+        left: 50%;
```
Comment (mab6bf) [inline src/_scss/components/visualizations/tooltip/_arrow.scss]: As discussed, `left: calc(50% - 8px);` should center the arrow

File: src/graphics/homepage.json [hunk 3]
```diff
@@ -67,5 +60,29 @@
...
+    "children": [
+        {"id": 0, "name": "Direct payments to individuals", "value": 1166902969415 },
+        {"id": 1, "name": "For Profit", "value": 719365549385 },
+        {"id": 2, "name": "Grants to state and local government", "value": 669721302076 },
+        {"id": 3, "name": "Non-profits; higher education", "value": 193757105484 },
+        {"id": 4, "name": "Running the government & direct services", "value": 1104353073640 }
     ]
+},
+"budgetDescriptions": [
+    {"id": 0, "name": "Direct payments to individuals", "value": "Lorem Ipsum stuff and things" },
```
Comment (ejgullo) [inline src/graphics/homepage.json]: 👍

File: src/js/components/homepage/visualizations/categoryMap/CategoryMap.jsx (hunk 1 above)
Comment (ejgullo) [inline src/js/components/homepage/visualizations/categoryMap/CategoryMap.jsx]: 👍

File: src/_scss/components/visualizations/tooltip/_arrow.scss (hunk 2 above)
Comment (ejgullo) [inline src/_scss/components/visualizations/tooltip/_arrow.scss]: 👍

**Comment by ejgullo** [comment]
Requested changes are available for review. :)

File: src/js/components/homepage/visualizations/categoryMap/CategoryMapCell.jsx [hunk 4]
```diff
@@ -105,6 +105,12 @@ export default class CategoryMapCell extends React.Component {
         if (height < 40 || width < 60) {
             percentView = 'none';
         }
+        let labelY = height / 2;
+        let percentY = (height / 2) + 20;
+        if (window.innerWidth < 480 && height < 60) {
```
Comment (mab6bf) [inline src/js/components/homepage/visualizations/categoryMap/CategoryMapCell.jsx]: This happens on the tablet width as well – can we change the innerWidth to 768?

**Pull request merged**